### PR TITLE
Adds pdf-lib and pdf cover page gen interactor  with test

### PR DESCRIPTION
**Pull request by matthopson**
- Adds `pdf-lib` library for PDF manipulation
- Adds `addCoverToPDFDocument` to interactors
- Adds `addCoverToPDFDocument` test

### CODE REVIEW

File: shared/src/business/useCases/addCoverToPDFDocumentInteractor.test.js [hunk 1]
```diff
@@ -0,0 +1,13308 @@
+const {
+  addCoverToPDFDocument,
+} = require('./addCoverToPDFDocumentInteractor.js');
+
+const { PDFDocumentFactory } = require('pdf-lib');
+
+function testPdfDocBytes() {
+  return new Uint8Array([
```
Comment (matthopson) [inline shared/src/business/useCases/addCoverToPDFDocumentInteractor.test.js]: Prettier didn't like this condensed. I assume we'll want to override that to make this a bit more readable, yeah? (I'll happily take the 13k line cred, tho.. haha)

Comment (ericsorenson) [reply shared/src/business/useCases/addCoverToPDFDocumentInteractor.test.js]: checking this out to take a look :-)

**Review by kkoskelin — APPROVED** [review]
Nifty.

File: shared/src/business/useCases/addCoverToPDFDocumentInteractor.test.js (hunk 1 above)
Comment (codyseibert) [reply shared/src/business/useCases/addCoverToPDFDocumentInteractor.test.js]: could we just put this array in a separate .json file and require it in?

**Comment by matthopson** [comment]
@ericsorenson @codyseibert created a dir in `shared` called `test-assets` with `sample.pdf` for use in any other tests requiring a sample pdf. Thoughts?

File: shared/src/business/useCases/addCoverToPDFDocumentInteractor.test.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+const fs = require('fs');
+const path = require('path');
+const {
+  addCoverToPDFDocument,
+} = require('./addCoverToPDFDocumentInteractor.js');
+
+const { PDFDocumentFactory } = require('pdf-lib');
+const testAssetsPath = path.join(__dirname, '../../../test-assets/');
```
Comment (codyseibert) [inline shared/src/business/useCases/addCoverToPDFDocumentInteractor.test.js:8]: 👍

**Pull request merged**